### ISSUE TITLE
util-script test-52 seems to hang when run in aggregate but passes when run by itself.

### ISSUE THREAD

**Issue by ndokos** [distributed-system-analysis/pbench#1792]
`tox -e util-scripts` seems to hang on test-52. OTOH `test -e util-scripts test-52` passes.

BTW, I interrupted the hanging test-52 with C-c, and had to kill -15 the tool-data-sink afterwards - else, python would raise an exception about `Address in use`.

**Comment by ndokos** [comment]
Here's the output after I interrupted it, just in case it helps:

```
test-52 test-start-stop-tool-meister mygroup
  C-c C-c*** /home/nick/src/github/pbench/worktrees/server-rpm/agent/util-scripts/gold/test-start-stop-tool-meister/test-52.txt	Mon Sep 14 11:38:55 2020
--- /var/tmp/pbench-test-utils/output.txt	Mon Sep 14 12:41:51 2020
***************
*** 6,12 ****
  waiting for all successfully spawned SSH processes to show up as subscribers
  channel payload, "'{"hostname": "testhost.example.com", "kind": "ds", "pid": NNNNN}'"
  channel payload, "'{"hostname": "testhost.example.com", "kind": "tm", "pid": NNNNN}'"
! --- Finished test-52 test-start-stop-tool-meister (status=0)
  +++ pbench tree state
  /var/tmp/pbench-test-utils/pbench
  /var/tmp/pbench-test-utils/pbench/mock-run
--- 6,19 ----
  waiting for all successfully spawned SSH processes to show up as subscribers
  channel payload, "'{"hostname": "testhost.example.com", "kind": "ds", "pid": NNNNN}'"
  channel payload, "'{"hostname": "testhost.example.com", "kind": "tm", "pid": NNNNN}'"
! Traceback (most recent call last):
!   File "/var/tmp/pbench-test-utils/opt/pbench-agent/util-scripts/pbench-tool-meister-stop", line 170, in <module>
!     status = main(sys.argv)
!   File "/var/tmp/pbench-test-utils/opt/pbench-agent/util-scripts/pbench-tool-meister-stop", line 131, in main
!     time.sleep(0.1)
! KeyboardInterrupt
! "pbench-tool-meister-stop" failed to execute successfully (exit code: 1)
! --- Finished test-52 test-start-stop-tool-meister (status=1)
  +++ pbench tree state
  /var/tmp/pbench-test-utils/pbench
  /var/tmp/pbench-test-utils/pbench/mock-run
***************
*** 14,21 ****
--- 21,30 ----
  /var/tmp/pbench-test-utils/pbench/mock-run/tm/pbench-tool-data-sink.err
  /var/tmp/pbench-test-utils/pbench/mock-run/tm/pbench-tool-data-sink.log
  /var/tmp/pbench-test-utils/pbench/mock-run/tm/pbench-tool-data-sink.out
+ /var/tmp/pbench-test-utils/pbench/mock-run/tm/pbench-tool-data-sink.pid
  /var/tmp/pbench-test-utils/pbench/mock-run/tm/redis.conf
  /var/tmp/pbench-test-utils/pbench/mock-run/tm/redis.log
+ /var/tmp/pbench-test-utils/pbench/mock-run/tm/redis_17001.pid
  /var/tmp/pbench-test-utils/pbench/mock-run/tm/tm-mygroup-testhost.example.com.err
  /var/tmp/pbench-test-utils/pbench/mock-run/tm/tm-mygroup-testhost.example.com.log
  /var/tmp/pbench-test-utils/pbench/mock-run/tm/tm-mygroup-testhost.example.com.out
***************
*** 31,49 ****
  [info][1900-01-01T00:00:00.000000] "mpstat" tool is now registered for host "testhost.example.com" in group "mygroup"
  --- pbench.log file contents
  +++ mock-run/tm/pbench-tool-data-sink.err file contents
- Bottle v#.##.## server starting up (using DataSinkWsgiServer(handler_class=<class '__main__.DataSinkWsgiServer.__init__.<locals>.DataSinkWsgiRequestHandler'>))...
- Listening on http://0.0.0.0:8080/
- Hit Ctrl-C to quit.
- 
  --- mock-run/tm/pbench-tool-data-sink.err file contents
  +++ mock-run/tm/pbench-tool-data-sink.log file contents
  INFO pbench-tool-data-sink run -- Running Bottle web server ...
  INFO pbench-tool-data-sink execute -- Terminate bottle server
- INFO pbench-tool-data-sink run -- Bottle web server exited
- INFO pbench-tool-data-sink main -- Remove pid file ... (pbench-tool-data-sink.pid)
  --- mock-run/tm/pbench-tool-data-sink.log file contents
  +++ mock-run/tm/pbench-tool-data-sink.out file contents
  --- mock-run/tm/pbench-tool-data-sink.out file contents
  +++ mock-run/tm/redis.conf file contents
  bind localhost
  daemonize yes
--- 40,54 ----
  [info][1900-01-01T00:00:00.000000] "mpstat" tool is now registered for host "testhost.example.com" in group "mygroup"
  --- pbench.log file contents
  +++ mock-run/tm/pbench-tool-data-sink.err file contents
  --- mock-run/tm/pbench-tool-data-sink.err file contents
  +++ mock-run/tm/pbench-tool-data-sink.log file contents
  INFO pbench-tool-data-sink run -- Running Bottle web server ...
  INFO pbench-tool-data-sink execute -- Terminate bottle server
  --- mock-run/tm/pbench-tool-data-sink.log file contents
  +++ mock-run/tm/pbench-tool-data-sink.out file contents
  --- mock-run/tm/pbench-tool-data-sink.out file contents
+ +++ mock-run/tm/pbench-tool-data-sink.pid file contents
+ 3811035--- mock-run/tm/pbench-tool-data-sink.pid file contents
  +++ mock-run/tm/redis.conf file contents
  bind localhost
  daemonize yes
***************
*** 58,67 ****
  # Configuration loaded
  # Server initialized
  * Ready to accept connections
- # User requested shutdown...
- * Removing the pid file.
- # Redis is now ready to exit, bye bye...
  --- mock-run/tm/redis.log file contents
  +++ mock-run/tm/tm-mygroup-testhost.example.com.err file contents
  --- mock-run/tm/tm-mygroup-testhost.example.com.err file contents
  +++ mock-run/tm/tm-mygroup-testhost.example.com.log file contents
--- 63,72 ----
  # Configuration loaded
  # Server initialized
  * Ready to accept connections
  --- mock-run/tm/redis.log file contents
+ +++ mock-run/tm/redis_17001.pid file contents
+ 3811029
+ --- mock-run/tm/redis_17001.pid file contents
  +++ mock-run/tm/tm-mygroup-testhost.example.com.err file contents
  --- mock-run/tm/tm-mygroup-testhost.example.com.err file contents
  +++ mock-run/tm/tm-mygroup-testhost.example.com.log file contents
FAIL - test-52
test-53 test-client-tool-meister 
ERROR: got KeyboardInterrupt signal
```

**Comment by portante** [comment]
Most likely previous `redis-server` processes, and / or `python3` test processes are present from a previously unsuccessful test.

Try making sure the environment is clean with, `killall redis-server python3`, repeating that command until it reports no pids for either one.

**Comment by ndokos** [comment]
Fixed by #1855
